src/UserSpaceInstrumentation/InstrumentProcess.cpp

```diff
@@ -103,6 +109,124 @@ bool IsBlocklisted(std::string_view function_name) {
   return kBlocklist.contains(function_name);
 }
 
+// MachineCodeForCloneCall creates the code to spawn a new thread inside the target process by using
+// the clone syscall. This thread is used to execute the initialization code inside the target.
+// Note that calling the result of the clone call a "thread" is a bit of a misnomer: We
+// do not create a new data structure for thread local storage but use the one of the thread we
+// halted.
+ErrorMessageOr<MachineCode> MachineCodeForCloneCall(pid_t pid, void* library_handle,
+                                                    uint64_t top_of_stack) {
+  constexpr uint64_t kCloneFlags =
+      CLONE_FILES | CLONE_FS | CLONE_IO | CLONE_SIGHAND | CLONE_SYSVSEM | CLONE_THREAD | CLONE_VM;
+  constexpr uint32_t kSyscallNumberClone = 0x38;
+  constexpr uint32_t kSyscallNumberExit = 0x3c;
+  constexpr const char* kInitializeInstrumentationFunctionName = "InitializeInstrumentation";
+  OUTCOME_TRY(void* initialize_instrumentation_function_address,
+              DlsymInTracee(pid, library_handle, kInitializeInstrumentationFunctionName));
+  MachineCode code;
+  code.AppendBytes({0x48, 0xbf})
+      .AppendImmediate64(kCloneFlags)  // mov kCloneFlags, rdi
```

Comment:
Sooo, sorry to mention this again, but now you have the order of the parameters as in the AT&T syntax, but the register names without `%` as in the Intel syntax...

Reply:
Yeah, right. Since I used something like Intel syntax in other files I'll change this to "dest, src" as well.